sdk/azidentity/azidentity.go

```diff
@@ -54,18 +54,19 @@ func (e *AADAuthenticationFailedError) Error() string {
 	return msg
 }
 
-// AuthenticationFailedError is a struct used to marshal responses when authentication has failed
+// AuthenticationFailedError is returned when the authentication request has failed.
 type AuthenticationFailedError struct {
 	inner error
 	msg   string
 }
 
-// Unwrap method on AuthenticationFailedError provides access to the inner error
+// Unwrap method on AuthenticationFailedError provides access to the inner error authentication
+// error returned from Azure Active Directory
 func (e *AuthenticationFailedError) Unwrap() error {
 	return e.inner
 }
 
-// IsNotRetriable allows retry policy to stop execution in case it receives a AuthenticationFailedError
+// IsNotRetriable allows retry policy to stop execution in case it receives an AuthenticationFailedError
 func (e *AuthenticationFailedError) IsNotRetriable() bool {
```

Comment:
We should change this to `Temporary()`, see https://golang.org/pkg/net/#Error for the pattern.

Reply:
Which also means it would return `false`.

Reply:
We can leave this discussion for a separate PR.

Reply:
Ok sounds good. I'll leave it as is for now, but I'm open to changing it to Temporary()